readthedocs/projects/models.py

```diff
@@ -1100,23 +1098,17 @@ def full_find(self, filename, version):
                 matches.append(os.path.join(root, match))
         return matches
 
+    @lru_cache(maxsize=1)
```

Comment:
this should also reduce the number of queries on .com, nothing like an old fashioned caching.